frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -1,110 +1,113 @@
-      </>
-    );
-  const uninstallMessage =
-    props.csv?.metadata?.annotations['operator.openshift.io/uninstall-message'];
+export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+  cancel,
+  close,
+  csv,
+  k8sKill,
+  subscription,
+}) => {
+  const { t } = useTranslation();
+  const [handlePromise, inProgress, errorMessage] = usePromiseHandler();
+  const submit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>): void => {
+      event.preventDefault();
+      const deleteOptions = {
+        kind: 'DeleteOptions',
+        apiVersion: 'v1',
+        propagationPolicy: 'Foreground',
+      };
+      handlePromise(
+        Promise.all([
+          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+          ...(subscription?.status?.installedCSV
+            ? [
+                k8sKill(
+                  ClusterServiceVersionModel,
+                  {
+                    metadata: {
+                      name: subscription.status.installedCSV,
+                      namespace: subscription.metadata.namespace,
+                    },
+                  },
+                  {},
+                  deleteOptions,
+                ).catch(() => Promise.resolve()),
+              ]
+            : []),
+        ]),
+      )
+        .then(() => {
+          close();
+          if (
+            window.location.pathname.split('/').includes(subscription.metadata.name) ||
+            window.location.pathname.split('/').includes(subscription.status.installedCSV)
+          ) {
+            history.push(
+              resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()),
+            );
+          }
+        })
+        .catch(() => {});
+    },
+    [close, handlePromise, k8sKill, subscription],
+  );
 
+  const name = csv?.spec?.displayName || subscription?.spec?.name;
+  const namespace =
+    subscription.metadata.namespace === GLOBAL_OPERATOR_NAMESPACE
+      ? 'all-namespaces'
+      : subscription.metadata.namespace;
+  const uninstallMessage = csv?.metadata?.annotations?.[OPERATOR_UNINSTALL_MESSAGE_ANNOTATION];
   return (
     <form onSubmit={submit} name="form" className="modal-content co-catalog-install-modal">
       <ModalTitle className="modal-header">
-        <YellowExclamationTriangleIcon className="co-icon-space-r" /> Uninstall Operator?
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> {t('olm~Uninstall Operator?')}
       </ModalTitle>
       <ModalBody>
-        This will remove operator <strong>{name}</strong> from {context}. Removing the operator will
-        not remove any of its custom resource definitions or managed resources. If your operator has
-        deployed applications on the cluster or configured off-cluster resources, these will
-        continue to run and need to be cleaned up manually.
+        <Trans t={t} ns="olm">
+          This will remove operator <strong>{{ name }}</strong> from{' '}
```

Comment:
Do we capitalize "Operator" elsewhere? We should try to be consistent whatever we choose.

Reply:
Yeah I think we do, I'll try to catch these wherever I see them